SpaceHosting.Index/Sparnn/Clusters/NonTerminalClusterIndex.cs

```diff
@@ -5,17 +5,19 @@
 using MoreLinq;
 using SpaceHosting.Index.Sparnn.Distances;
 using SpaceHosting.Index.Sparnn.Helpers;
+using MSparseVector = MathNet.Numerics.LinearAlgebra.Double.SparseVector;
 
 namespace SpaceHosting.Index.Sparnn.Clusters
 {
     internal sealed class NonTerminalClusterIndex<TRecord> : BaseClusterIndex<TRecord>
         where TRecord : notnull
     {
         private readonly MatrixMetricSearchSpaceFactory matrixMetricSearchSpaceFactory;
-        private MatrixMetricSearchSpace<IClusterIndex<TRecord>> clusterSpace = null!;
+        private IMatrixMetricSearchSpace<IClusterIndex<TRecord>> clusterSpace = null!;
+        private Random random => new Random(42);
```

Comment:
We should  not use fixed seed for all `NonTerminalClusterIndex` instances. The idea was to add _ability_ to fixate it in tests only. So `Random` instance should be passed here as parameter.